license-maven-plugin-git/src/main/java/com/mycila/maven/plugin/license/git/GitLookup.java

```diff
@@ -76,7 +77,8 @@ public GitLookup(File anyFile, DateSource dateSource, TimeZone timeZone, int che
     super();
     this.repository = new FileRepositoryBuilder().findGitDir(anyFile).build();
     /* A workaround for  https://bugs.eclipse.org/bugs/show_bug.cgi?id=457961 */
-    this.repository.getObjectDatabase().newReader().getShallowCommits();
+    // Also contains contents of .git/shallow and can detect shallow repo
+    this.shallow = !this.repository.getObjectDatabase().newReader().getShallowCommits().isEmpty();
```

Comment:
This reader was not closed here... We missed a try/resource I think ?

Reply:
Yeah, I saw the warning in my IDE, but everywhere else I see this workaround posted, nobody is using try with resources and it's not obvious to me that it needs to be closed, and possibly closing it would negate whatever the workaround is.  Given my earlier mistakes on resources I decided to leave it as-is (other than collecting some side-effect information from it).